### PR TITLE
refac: use new context for reading database values

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

To ensure we are not just reading from memory in the old context, its been decided to create a new database context for the assert part of database tests

ACs
1. The context used to arrange and act is not the same as the one used to assert that data is valid

## References

* #555

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeRepositoryTests.cs [hunk 1]
```diff
@@ -50,62 +51,76 @@ public ChargeRepositoryTest(SqlServerResource<SqlServerOptions> resource)
...
+            await using var chargesDatabaseReadContext = await SquadronContextFactory
+                .GetDatabaseContextAsync(_resource)
+                .ConfigureAwait(false);
+            var expected = await chargesDatabaseReadContext.Charges.SingleAsync(x =>
+                x.ChargeId == charge.Id &&
+                x.MarketParticipant.MarketParticipantId == charge.Owner &&
+                x.ChargeType == (int)charge.Type)
+                .ConfigureAwait(false);
+
             Assert.NotNull(expected);
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeRepositoryTests.cs:74]:
```suggestion
            var actual = await chargesDatabaseReadContext.Charges.SingleAsync(x =>
                x.ChargeId == charge.Id &&
                x.MarketParticipant.MarketParticipantId == charge.Owner &&
                x.ChargeType == (int)charge.Type)
                .ConfigureAwait(false);

            Assert.NotNull(actual);
```

Comment (BjarkeMeier) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeRepositoryTests.cs]: This "expected" vs "actual" naming fix is needed in multiple places.

**Review by BjarkeMeier — APPROVED** [review]
LGTM. It would be nice it you'll fix the expected/actual naming as you're touching the code - though I know they were not introduced in this PR.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeRepositoryTests.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Infrastructure/Repositories/ChargeRepositoryTests.cs]: fixed

**Review by HenrikSommer — APPROVED** [review]

**Comment by BjarkeMeier** [comment]
CodeCov report doesn't look good. Why are coverage of all those files massively reduced?

**Pull request merged**